syntax = "proto2";

package Proto;

message AccountCheck {
	required string account = 1;
	required string password = 2;
}

message AccountCheckRs {
	enum ReturnCode {
		ARC_OK = 0;
		ARC_UNKONWN = 1;
		ARC_NOT_FOUND_ACCOUNT = 2;
		ARC_PASSWORD_WRONG = 3;
		ARC_LOGGING = 4; // 正在登录
		ARC_TIMEOUT = 5; // 超时
	}

	required int32 return_code = 1;
}

message AccountCheckToHttpRs {
	required int32 return_code = 1;	// AccountCheckRs::ReturnCode
	required string account = 2;
}

message RobotState {
	required string account = 1;
	required int32 state = 2;
}

message RobotSyncState {
	repeated RobotState states = 1;
}

message RobotTestEnd {
	required int32 state = 1;
}